src/fil_cns/validation.rs

```diff
@@ -127,19 +127,17 @@ pub(in crate::fil_cns) async fn validate_block<DB: Blockstore + Sync + Send + 's
 
     // Beacon values check
     if std::env::var(IGNORE_DRAND_VAR) != Ok("1".to_owned()) {
-        let v_block = Arc::clone(&block);
-        let parent_epoch = base_tipset.epoch();
-        let v_prev_beacon = Arc::clone(&prev_beacon);
-        validations.push(tokio::task::spawn(async move {
-            v_block
-                .header()
-                .validate_block_drand(
-                    win_p_nv,
-                    beacon_schedule.as_ref(),
-                    parent_epoch,
-                    &v_prev_beacon,
-                )
-                .map_err(|e| FilecoinConsensusError::BeaconValidation(e.to_string()))
+        validations.push(tokio::task::spawn({
+            let block = Arc::clone(&block);
+            let parent_epoch = base_tipset.epoch();
+            let prev_beacon = Arc::clone(&prev_beacon);
+            let nv = state_manager.get_network_version(header.epoch);
```

Comment:
This is a bug, it might cause failure when nv < V15
Previously in forest:
`let win_p_nv = state_manager.get_network_version(parent_epoch)`

But in lotus (See https://github.com/filecoin-project/lotus/blob/e68b762b40cd650410d5b203c578e196f32427d9/chain/consensus/filcns/filecoin.go#L246C3-L249C4)
```go
nv := filec.sm.GetNetworkVersion(ctx, h.Height)
		if err := beacon.ValidateBlockValues(filec.beacon, nv, h, baseTs.Height(), *prevBeacon); err != nil {
			return xerrors.Errorf("failed to validate blocks random beacon values: %w", err)
		}
```